### PR TITLE
[test-only] ApiTest.check all response items 

### PR DESCRIPTION
**Pull request by ScharfViktor**
Idea: 
- Create api tests as contract tests. 
- Tests should check the status of the response and check elements in the body of the response

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/search.feature [hunk 1]
```diff
@@ -121,7 +120,6 @@ Feature: Search
     And the search result should contain "1" entries
     And the search result of user "Brian" should contain these entries:
       | /<searchObject> |
-    And for user "Brian" the response should contains the parent "<parentFolder>" from mountpoint "folderMain"
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/search.feature:124]: don't we need that check anymore?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -2995,19 +2995,41 @@ public function searchResultShouldContainSpace(string $user, string $spaceName):
 	}
 
 	/**
-	 * @Then /^for user "([^"]*)" the response should contains the parent "([^"]*)" from (?:space|mountpoint) "([^"]*)"$/
+	 * @Then /^for user "([^"]*)" the REPORT response should contains a (?:space|mountpoint) "([^"]*)" with these key and value pairs:$/
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2998]:
```suggestion
	 * @Then /^for user "([^"]*)" the REPORT response should contain a (?:space|mountpoint) "([^"]*)" with these key and value pairs:$/
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -2995,19 +2995,41 @@ public function searchResultShouldContainSpace(string $user, string $spaceName):
...
 	 */
-	public function responseShouldContainParent(string $user, string $parent, string $space): void {
-		// get a response after a Report request (called in the core)
-		$responseArray = json_decode(json_encode($this->featureContext->getResponseXml()->xpath("//d:response/d:propstat/d:prop/oc:file-parent")), true, 512, JSON_THROW_ON_ERROR);
-		Assert::assertNotEmpty($responseArray, "search result is empty");
-		Assert::assertEquals($this->getFolderId($user, $space, $parent), $responseArray[0][0], 'wrong file-parentId');
+	public function reportResponseShouldContain(string $user, string $space, TableNode $table): void {
+		$this->featureContext->verifyTableNodeColumns($table, ['key', 'value']);
+		$xmlRes = $this->featureContext->getResponseXml();
+		$resourceType = json_decode(json_encode($xmlRes->xpath("//d:response/d:propstat/d:prop/d:getcontenttype")), true, 512, JSON_THROW_ON_ERROR);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3010]: I think something like this also should work:
```suggestion
		$resourceType = $xmlRes->xpath("//d:response/d:propstat/d:prop/d:getcontenttype")[0]->__toString();
```

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Web-Tests-ocis-smoke-ocis-storage-1</strong>](https://drone.owncloud.com/owncloud/ocis/16398/58/1) failed. Further test are cancelled...

File: tests/acceptance/features/apiSpaces/search.feature (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpaces/search.feature:124]: I deleted case. It is duplicate with `report.feature`

File: tests/acceptance/features/bootstrap/SpacesContext.php (hunk 3 above)
Comment (ScharfViktor) [reply tests/acceptance/features/bootstrap/SpacesContext.php:3010]: Thanks, It works

**Review by individual-it — APPROVED** [review]

**Pull request merged**